### PR TITLE
What you wear and on what chair you sit on can now influence fishing difficulty

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
A foreword, I had to refactor a few bits of shitcode my past self added first. For context, the "gone fishing" and "actively fishing" traits only had one source, which is the fishing challenge itself, ad there was no way to access the challenge from outside its code, except for a few weakrefs which were being used as sources for the aforementioned traits (the shitcode in a nutshell). There were also a few signals that I didn't like because they were being sent to the harder-to-access challenge datum rather than the user. So I scrapped the traits for a couple signals to send to the user, then added a global list as a mean to easily access the challenge datum, and lastly changed the code to accomodate the titled feature (and allow the challenge to recalculate its difficulty DURING the minigame phase) 

Moving on to the actual feature: I've added a component that can be added to objects on which mobs can be buckled to or items. When equipped in the right slots or buckled to, the object will adjust the difficulty of current and future fishing challenges by a certain amount (more often than not positive, but there're many exceptions) as long as the object isn't equipped or the user is unbuckled.

I've been having some fun adding component to a ton of clothes in the game as well as chairs. Way too many objects to enumerate, so I'll give you the general idea:
- each carp-themed article provides a slight positive modifier (easier)
- some (not all) doctor-related garbs provide a marginal positive modifier each (fish doctor jokes)
- floortile camo clothes have positive modifiers
- Tuxedo, laceups, gowns provide negative modifier (more difficult)
- utility garbs such as bio/bomb/rad hoods and suits are quite bad. Riot armor too.
- boxing gloves are very, very bad. Insulated gloves and haul gauntlets are also very bad, to a lesser degree.
- **tackle** gloves are good. (pun intended)
- wizard garbs are good, because wizards are good at casting. (also a pun)
- magboots slightly bad. Space suits bad.
- Blindfolds and welding protection are also bad. Gas masks marginally bad.
- Pirate attire is nice to have. (I just vibed a little on this one)
- plastic chairs are quite versatile because they can be carried around, but the mime chair is the best, followed by ratvarian chairs.
- Fishing toolboxes, analyzers and the fish catalog are a plus, because they can be held.
- And the fishing hat, obviously (not as great as you'd think)

Some of these may be subject to change depending on what people say.

## Why It's Good For The Game
A hundred lines of fishing challenge code made ever-so-slightly less awful, and a way to modify fishing diffculty beside skills and bait.

## Changelog

:cl:
add: Your current clothes and what chair you sit on can now influence the difficulty of fishing minigames. Having a bare minimum of fishing skill will let you distinguish which objects can help and which won't, so keep an eye out. Holding fishing toolboxes, fish analyzers or fish catalogs can also help.
/:cl:

### CODE REVIEW

**Comment by Loafin34** [comment]
at this rate we need a new material called fishtonium

**Comment by DrAmazing343** [comment]
mimeta too strong... in all truth I fucking adore this pr because it is the essence of what most people call soul. it is the little tiny interactions and entirely too complex system nested within a feature and all of it oozing with puns, witty irony, and a little bit of necessary practicality to play foil to both of those.

Thank you Ghommie!!

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

**Comment by Loafin34** [comment]
can you add in a thing for the mime chair that says something like "summons a luxuriously comfortable chair. You get a feeling of piscine superiority emanating from it."

File: code/modules/mod/modules/modules_general.dm [hunk 1]
```diff
@@ -1015,15 +1015,17 @@
 			qdel(gloves.GetComponent(/datum/component/profound_fisher))
 
 /obj/item/mod/module/fishing_glove/on_suit_activation()
+	var/obj/item/gloves = mod.get_part_from_slot(ITEM_SLOT_GLOVES)
+	gloves.AddComponent(/datum/component/adjust_fishing_difficulty, 5)
 	if(!equipped)
 		return
-	var/obj/item/gloves = mod.get_part_from_slot(ITEM_SLOT_GLOVES)
 	if(gloves)
 		gloves.AddComponent(/datum/component/profound_fisher, equipped)
```
Comment (ninjanomnom) [inline code/modules/mod/modules/modules_general.dm:1023]: shouldn't it be with the other AddComponent if gloves can be null?

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

**Review by ninjanomnom — APPROVED** [review]

**Pull request merged**